tests/istanbul/eip152_blake2/test_blake2.py

```diff
@@ -0,0 +1,547 @@
+            ),
+            id="EIP-192-0016-120",
+        ),
+        pytest.param(
+            Blake2Input(
+                rounds=32,
+                h="48c9bdf267e6096a3ba7ca8485ae67bb2bf894fe72f36e3cf1361d5f3af54fa5d182e6ad7f520e511f6c3e2b8c68059b6bbd41fbabd9831f79217e1319cde05b",
+                m="6162636465666768696a6b6c6d6e6f707172737475767778797a7b7c7d7e7f808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9fa0a1a2a3a4a5a6a7a8a9aaabacadaeafb0b1b2b3b4b5b6b7b8b9babbbcbdbebfc0c1c2c3c4c5c6c7c8c9cacbcccdcecfd0d1d2d3d4d5d6d7d8d900000000000000",
+                t_0="7800000000000000",
+                t_1="0000000000000000",
+            ),
+            ExpectedOutput(
+                call_return_code="0x01",
+                data_1="0x39fc2077154fba422b3d628d10908c596beebea8dfd90f14566aec4f60bdb2bc",
+                data_2="0xa75d73ab2b224d58c3568cbc7fc8905cc849f10745f00addef02384032d53729",
+            ),
+            id="EIP-192-0032-120",
+        ),
+        pytest.param(
+            Blake2Input(
+                rounds=64,
+                h="48c9bdf267e6096a3ba7ca8485ae67bb2bf894fe72f36e3cf1361d5f3af54fa5d182e6ad7f520e511f6c3e2b8c68059b6bbd41fbabd9831f79217e1319cde05b",
+                m="6162636465666768696a6b6c6d6e6f707172737475767778797a7b7c7d7e7f808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9fa0a1a2a3a4a5a6a7a8a9aaabacadaeafb0b1b2b3b4b5b6b7b8b9babbbcbdbebfc0c1c2c3c4c5c6c7c8c9cacbcccdcecfd0d1d2d3d4d5d6d7d8d900000000000000",
+                t_0="7800000000000000",
+                t_1="0000000000000000",
+            ),
+            ExpectedOutput(
+                call_return_code="0x01",
+                data_1="0x5bb981381beb687d5fdbe5e7c096fbd1ce193b780948c1d74ebbb7c58db364c7",
+                data_2="0xb7695d32f918444dbdcbdcff476fc70a926e228c4cbb7d05473711d3b56e5b33",
+            ),
+            id="EIP-192-0064-120",
+        ),
+    ],
+)
+def test_blake2(
+    state_test: StateTestFiller,
+    data: Blake2Input,
+    output: ExpectedOutput,
+    pre: Alloc,
+):
+    """Test BLAKE2 precompile."""
+    env = Environment()
+
+    account = pre.deploy_contract(
+        # Store all CALLDATA into memory (offset 0)
+        Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE())
+        # Store the returned CALL status (success = 1, fail = 0) into slot 0:
+        + Op.SSTORE(
+            0,
+            # Setup stack to CALL into Blake2 with the CALLDATA and CALL into it (+ pop value)
+            Op.CALL(Op.GAS(), 9, 0, 0, Op.CALLDATASIZE(), 0x200, 0x40),
+        )
+        + Op.SSTORE(
+            1,
+            Op.MLOAD(0x200),
+        )
+        + Op.SSTORE(
+            2,
+            Op.MLOAD(0x220),
+        )
+        + Op.STOP(),
+    )
+    sender = pre.fund_eoa()
+
+    tx = Transaction(
+        ty=0x0,
+        to=account,
+        data=data.create_blake2_tx_data(),
+        gas_limit="0xF000000000",
```

Comment:
This is certainly an extreme amount of gas 😅 

```suggestion
        gas_limit=1_000_000,
```

I tested locally and all cases pass with this reduction, except `case8`.

It seems like case 8 round count and 0xF000000000 as gas combined result in the EELS server hanging indefinitely and all subsequent tests fail because of this.

I think it should be ok to expect that the case gracefully runs out of gas if this many rounds are requested.